openghg/store/base/_datasource.py

```diff
@@ -23,13 +23,17 @@ class Datasource:
     _datasource_root = "datasource"
     _data_root = "data"
 
-    def __init__(self) -> None:
+    def __init__(self, uuid: Optional[str] = None) -> None:
         from collections import defaultdict
         from uuid import uuid4
 
         from openghg.util import timestamp_now
 
-        self._uuid: str = str(uuid4())
+        if uuid is None:
```

Comment:
This was actually causing issues with one of the mocked tests. What was happening originally was a new uuid was being created using `uuid()` even when the `.load(uuid=uuid)` method was being called - this still uses the __init__ with the `d = cls()` line. Currently, the attributes are being populated (new uuid being generated and then added) and then overwritten within the `.load()` method. The mocker was creating a limited set of uuids for use and so this was being unnecessarily exhausted due to the extra (and not needed) calls to `uuid().

So I added this as a quick way to stop `uuid()` being called again every time the Datasource was accessed (not just created). I added a note in the `load()` method about maybe moving out these attribute definitions in the long term to stop this being necessary.

Reply:
It was the `tests/store/test_emissions.py::test_read_binary_data()` which picked this up by the way.

Reply:
OK yeah, I think we can move to use the constructor more instead of the loading and saving functions. This is what I've done with the work for the multiple object stores.